### ISSUE TITLE
Test PnP.Powershell 2.5.0 to see if it has the fields currently missing from our Sharepoint provider

### ISSUE THREAD

**Issue by tkol2022** [cisagov/ScubaGear#1231]
# 💡 Summary #

Currently, the ScubaGear Sharepoint provider is not able to get the fields below because they are not provided by the underlying  PnP.Powershell module. This prevents ScubaGear from being able to assess a couple of Sharepoint / OneDrive policies. The PnP.Powershell module that ScubaGear is currently using is 1.12.0 which is old - the reason we are tied to that is because the newer modules require Powershell 7.

The scope of this issue is to perform a hands-on test the items below and then use that information to make decisions for ScubaGear:

- Try running ScubaGear with the latest version of PnP.Powershell which is 2.5.0
- Examine if PnP.Powershell 2.5.0 contains the fields which are currently missing

Currently missing fields in ScubaGear:

- GuestSharingGroupAllowListInTenantByPrincipalIdentity
- OnedriveSharingCapability

![image](https://github.com/user-attachments/assets/693dcd88-c258-4648-82a6-edbc88d1ef17)

## Motivation and context ##

We need to determine a strategy for getting these missing fields. If the fields are there in the latest PnP.Powershell, we know that we can get them when we migrate ScubaGear to Powershell 7 since at that point we will be able to use the latest version of PnP.Powershell.

## Implementation notes ##

- [x] Test the items above and document the results in this issue
- [ ] Keep the issue open for a couple of weeks to solicit feedback from team members

**Comment by tkol2022** [comment]
I tested ScubaGear with the latest version of PnP.Powershell which is 2.5.0. I confirmed that the latest PnP.Powershell will not work with Powershell 5.1 at all. It immediately errors out with a descriptive message explaining the dependency on Powershell 7.

**Comment by tkol2022** [comment]
I installed Powershell 7 separately of ScubaGear and then ran the Get-PnPTenant cmdlet. It does NOT retrieve the missing fields documented above. I filed an [issue in the PnP GitHub](https://github.com/pnp/powershell/issues/4111). 

In the future if we migrate ScubaGear to Powershell 7, we would only get the missing fields if the PnP dev team adds them.

As of now, the only off-the-shelf project I was able to find that [retrieves those fields is named PnP Core SDK](https://pnp.github.io/pnpcore/api/PnP.Core.Admin.Model.SharePoint.ITenantProperties.html) which is for .NET so it's not clear if ScubaGear could directly leverage it.
